rest/src/main/java/org/jboss/pnc/rest/notifications/model/NotificationFactory.java

```diff
@@ -0,0 +1,34 @@
+package org.jboss.pnc.rest.notifications.model;
+
+import org.jboss.pnc.spi.BuildStatus;
+import org.jboss.pnc.spi.events.BuildStatusChangedEvent;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.HashMap;
+import java.util.Map;
+
+@ApplicationScoped
+public class NotificationFactory {
+
+    private Map<BuildStatus, NotificationEventType> externalEvents = new HashMap<>();
+
+    public NotificationFactory() {
+        externalEvents.put(BuildStatus.REPO_SETTING_UP, NotificationEventType.BUILD_STARTED);
```

Comment:
Some other "clients" might want so show more progress details. I find it useful to see that my task is eg. COLLECTING_RESULTS_* or that task failed due to BUILD_ENV_SETUP and not because of the BUILD_COMPLETED_WITH_ERROR. Or maybe "WAITING_FOR_DEPENDENCIES" tells that the particular component is still waiting to be scheduled. Don't you think we should show this to the users ?
And org.jboss.pnc.spi.BuildStatus is already a part of the SPI.